SYCL/ESIMD/api/functional/ctors/common.hpp

```diff
@@ -29,54 +29,61 @@ using shared_allocator = sycl::usm_allocator<DataT, sycl::usm::alloc::shared>;
 template <typename DataT>
 using shared_vector = std::vector<DataT, shared_allocator<DataT>>;
 
-// Calls simd constructor in provided invocation context, which depends on the
-// TestCaseT type. TestCaseT is a struct, that should have call_simd_ctor method
-// that should return constructed object of simd class.
-// This function returns std::vector instance with the output data.
-template <typename DataT, int NumElems, typename TestCaseT>
-auto call_simd(sycl::queue &queue, const shared_vector<DataT> &ref_data) {
-
-  shared_vector<DataT> result{NumElems, shared_allocator<DataT>{queue}};
-
-  queue.submit([&](sycl::handler &cgh) {
-    const auto ref = ref_data.data();
-    auto out = result.data();
+template <typename DataT, int NumElems = 0>
+class TestDescription : public ITestDescription {
+protected:
+  const std::string m_data_type;
+  const std::string m_test_case_description;
+  DataT m_retrieved_val;
+  DataT m_expected_val;
+  size_t m_index;
+
+public:
+  TestDescription(const std::string &test_case_description,
+                  const std::string &data_type, DataT retrieved_val,
+                  DataT expected_val, size_t index)
+      : m_data_type(data_type), m_test_case_description(test_case_description),
+        m_retrieved_val(retrieved_val), m_expected_val(expected_val),
```

Comment:
```suggestion
  TestDescription(const std::string &data_type, DataT retrieved_val,
                  DataT expected_val, size_t index)
      : m_data_type(data_type), m_retrieved_val(retrieved_val), 
        m_expected_val(expected_val),
```
We would need no `get_description()` call on the upper level if we add the `typename ContextT` template parameter to the struct itself. Do you have any concerns?

Also, just minor suggestion, feel free to ignore, but what would you say to have `TestDescription(i, retrieved, expected, type_name)` sequence for the call parameters? So we might have type name always on the last place (the same way it's required for `for_all_types`, so we might just make it consistent over the codebase).

Reply:
This comment was applied [here](https://github.com/intel/llvm-test-suite/pull/614/commits/f5fcba9f39f6f63e6e4c8d2de3b4ea8cc64a0444#diff-a1a30a06ba7da50ce8e13a9af9caec10d385910429792a1de956a7f15cc3d646R40-R45)